### ISSUE TITLE
Add retry/resume to internal::ConnectionImpl::PartitionQuery

### ISSUE THREAD

**Issue by coryan** [googleapis/google-cloud-cpp-spanner#642]
We need to:

- [ ] Implement the functionality.
- [ ] Write a unit test verifying that it tolerates at least one transient error.
- [ ] Write a unit test verifying that it stops on the first permanent error.
- [ ] Write a unit test verifying that it stops when there are too many transient failures.

**Comment by coryan** [comment]
This is probably blocked by #271.

**Comment by scotthart** [comment]
#646 Implemented retry/resume for internal::ConnectionImpl::PartitionQuery
Client::ExecuteSql(QueryPartition) is handled by internal::ConnectionImpl::ExecuteSql(ExecuteSqlParams) so this issue is a duplicate of #643

**Comment by coryan** [comment]
Shouldn't we have unit tests in case we refactor this code differently in the future?

**Comment by coryan** [comment]
I was confused, the tests already exist for `internal::ConnectionImpl`.
